File: librococoa/src/main/java/org/rococoa/cocoa/foundation/NSNotificationCenter.java
/*
 * Copyright 2007, 2008 Duncan McGregor
 *
 * This file is part of Rococoa, a library to allow Java to talk to Cocoa.
 *
 * Rococoa is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Rococoa is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with Rococoa.  If not, see <http://www.gnu.org/licenses/>.
 */

package org.rococoa.cocoa.foundation;

import org.rococoa.ID;
import org.rococoa.ObjCClass;
import org.rococoa.Rococoa;
import org.rococoa.Selector;


public abstract class NSNotificationCenter extends NSObject {
    public static final _Class CLASS = Rococoa.createClass("NSNotificationCenter", _Class.class); //$NON-NLS-1$

    public interface _Class extends ObjCClass {
        NSNotificationCenter defaultCenter();
    }

    public abstract void addObserver_selector_name_object(ID notificationObserver,
                                                          Selector notificationSelector,
                                                          String notificationName,
                                                          NSObject notificationSender);

    public abstract void removeObserver(ID notificationObserver);

    public abstract void postNotification(NSNotification notification);

}

File: librococoa/src/main/kotlin/darwin/NSNotificationCenter.kt
package darwin 

import org.rococoa.ID
import org.rococoa.ObjCClass
import org.rococoa.Selector

/// <i>native declaration : :29</i>
abstract class NSNotificationCenter : NSObject() {
	interface _Class : ObjCClass {
		/**
		 * Original signature : `defaultCenter()`<br></br>
		 * *native declaration : :36*
		 */
		open fun defaultCenter(): NSNotificationCenter?
	}

	fun addObserver(
		notificationObserver: ID?,
		notificationSelector: Selector?,
		notificationName: String?,
		notificationSender: ID?
	) {
		this.addObserver_selector_name_object(
			notificationObserver,
			notificationSelector,
			notificationName,
			notificationSender
		)
	}

	/**
	 * *native declaration : :38*<br></br>
	 * Conversion Error : /// Original signature : `void addObserver(null, null, NSString*, null)`<br></br>
	 * - (void)addObserver:(null)observer selector:(null)aSelector name:(NSString*)aName object:(null)anObject; (Argument observer cannot be converted)
	 */
	abstract fun addObserver_selector_name_object(
		notificationObserver: ID?,
		notificationSelector: Selector?,
		notificationName: String?,
		notificationSender: ID?
	)

	/**
	 * Original signature : `void postNotification(NSNotification*)`<br></br>
	 * *native declaration : :40*
	 */
	abstract fun postNotification(notification: NSNotification?)
	/**
	 * *native declaration : :41*<br></br>
	 * Conversion Error : /// Original signature : `void postNotificationName(NSString*, null)`<br></br>
	 * - (void)postNotificationName:(NSString*)aName object:(null)anObject; (Argument anObject cannot be converted)
	 */
	/**
	 * *native declaration : :42*<br></br>
	 * Conversion Error : /// Original signature : `void postNotificationName(NSString*, null, NSDictionary*)`<br></br>
	 * - (void)postNotificationName:(NSString*)aName object:(null)anObject userInfo:(NSDictionary*)aUserInfo; (Argument anObject cannot be converted)
	 */
	/**
	 * *native declaration : :44*<br></br>
	 * Conversion Error : /// Original signature : `void removeObserver(null)`<br></br>
	 * - (void)removeObserver:(null)observer; (Argument observer cannot be converted)
	 */
	abstract fun removeObserver(notificationObserver: ID?)

	/**
	 * *native declaration : :45*<br></br>
	 * Conversion Error : /// Original signature : `void removeObserver(null, NSString*, null)`<br></br>
	 * - (void)removeObserver:(null)observer name:(NSString*)aName object:(null)anObject; (Argument observer cannot be converted)
	 */
	companion object {
		private val CLASS: _Class = org.rococoa.Rococoa.createClass("NSNotificationCenter", _Class::class.java)

		fun defaultCenter(): NSNotificationCenter? {
			return CLASS.defaultCenter()
		}
	}
}
